admiral/pkg/clusters/serviceentry.go

```diff
@@ -604,6 +628,122 @@ func AddServiceEntriesWithDr(ctx context.Context, rr *RemoteRegistry, sourceClus
 	}
 }
 
+// deleteAdditionalEndpoints deletes all the additional endpoints that were generated for this
+// ServiceEntry.
+func deleteAdditionalEndpoints(ctx context.Context, serviceEntry *networking.ServiceEntry, namespace string, rc *RemoteController) error {
+
+	additionalEndpointSuffixes := common.GetAdditionalEndpointSuffixes()
+	if len(additionalEndpointSuffixes) <= 0 {
+		log.Debugf("no additional endpoints to delete as no additional endpoints suffixes were found")
+		return nil
+	}
+
+	if serviceEntry == nil {
+		return fmt.Errorf("failed deleting additional endpoints for serviceentry as service entry object passed is nil")
+	}
+	if serviceEntry.Hosts == nil {
+		return fmt.Errorf("failed deleting additional endpoints for serviceentry as service entry hosts is nil")
+	}
+	if len(serviceEntry.Hosts) <= 0 {
+		return fmt.Errorf("failed deleting additional endpoints for serviceentry as service entry hosts is empty")
+	}
+
+	destinationHostName := serviceEntry.Hosts[0]
+
+	virtualServiceHostnames := replaceEndpointSuffix(destinationHostName, additionalEndpointSuffixes)
+	if len(virtualServiceHostnames) <= 0 {
+		return fmt.Errorf("failed deleting additional endpoints for serviceentry for SE host: '%s' and additional endpoint suffixes %s", destinationHostName, additionalEndpointSuffixes)
+	}
+
+	defaultVSName := getIstioResourceName(virtualServiceHostnames[0], "-vs")
+	vsToDelete := &v1alpha3.VirtualService{
+		ObjectMeta: v12.ObjectMeta{
+			Name: defaultVSName,
+		},
+	}
+	err := deleteVirtualService(ctx, vsToDelete, namespace, rc)
+	if err != nil {
+		log.Infof(LogErrFormat, "Delete", "VirtualService", defaultVSName, rc.ClusterID, err)
+		return err
+	}
+	log.Infof(LogFormat, "Delete", "VirtualService", defaultVSName, rc.ClusterID, "Success")
+	return nil
+}
+
+// createAdditionalEndpoints creates additional endpoints of service defined in the ServiceEntry.
+// The list suffixes defined in admiralparams.AdditionalEndpointSuffixes will used to generate the endpoints
+func createAdditionalEndpoints(ctx context.Context, serviceEntry *networking.ServiceEntry, namespace string, rc *RemoteController) error {
+
+	additionalEndpointSuffixes := common.GetAdditionalEndpointSuffixes()
```

Comment:
Both create and delete are using common functionality, may be abstract them in a function so both can reuse it.